src/Core/Ast/Parser/NikicPhpParser/FileReferenceVisitor.php

```diff
@@ -229,4 +233,52 @@ private function leaveGroupUse(GroupUse $node): void
             }
         }
     }
+
+    private function processClassDocs(ClassLike $node): void
+    {
+        $docComment = $node->getDocComment();
+        if (!$docComment instanceof Doc) {
+            return;
+        }
+
+        $tokens = new TokenIterator($this->lexer->tokenize($docComment->getText()));
+        $docNode = $this->docParser->parse($tokens);
+
+        foreach ($docNode->getMethodTagValues() as $methodTagValue) {
+            $templateTypes = array_merge(
```

Comment:
Would the spread operator improve the performance?

Reply:
No idea. The alternative I came up with using spread operator:
```php
            $templateTypes = [
                ...array_map(
                    static fn (TemplateTagValueNode $node): string => $node->name,
                    $methodTagValue->templateTypes
                ),
                ...$this->currentReference->getTokenTemplates()
            ];
```
is using it twice and just seems more confusing to me than the current implementation. But I would be happy to adopt it if we have evidence it helps performance bottlenecks.